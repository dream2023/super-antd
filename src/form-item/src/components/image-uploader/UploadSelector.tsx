import type { ProFieldProps } from '@ant-design/pro-utils/lib/typing';
import type { FC } from 'react';
import React from 'react';

export interface UploadSelectorProps {
  /**
   * 上传占位的 icon
   *
   * @default `<UploadOutlined />`
   */
  icon?: React.ReactNode;
  /**
   * 上传占位标题
   */
  title?: React.ReactNode;
  // 上传最大数量
  max: number;
  // 是否多选
  multiple?: boolean;
  // 文件
  filesCount: number;
  // 是否为只读模式
  mode?: ProFieldProps['mode'];
  // 预览图限制
  previewStyle?: { width: number; height: number };
}

const UploadSelector: FC<UploadSelectorProps> = ({
  icon,
  title,
  max,
  multiple,
  filesCount,
  mode,
  previewStyle = {},
}) => {
  // 是否显示上传
  const showUpload = (multiple || filesCount < max) && mode !== 'read';
  if (!showUpload) return null;

  return (
    <div className="super-antd-center" style={previewStyle}>
      <div>
        <div>{icon}</div>
        <div>{title}</div>
      </div>
    </div>
  );
};

export default UploadSelector;
