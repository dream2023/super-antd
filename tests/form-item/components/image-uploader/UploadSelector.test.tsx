import { render } from '@testing-library/react';
import React from 'react';

import UploadSelector from '@/form-item/src/components/image-uploader/UploadSelector';

describe('UploadSelector', () => {
  test('最大数量不大于 max 时，则不渲染内容', () => {
    const wrapper = render(<UploadSelector max={2} filesCount={2} />);
    expect(wrapper.container.innerHTML).toBe('');
  });

  test('当 multiple 为 true 时，渲染内容', () => {
    const wrapper = render(<UploadSelector multiple max={1} filesCount={2} />);
    expect(wrapper.container.innerHTML).not.toBe('');
  });

  test('当为 mode 为 read 时，则不渲染内容', () => {
    const wrapper = render(<UploadSelector max={2} filesCount={1} mode="read" />);
    expect(wrapper.container.innerHTML).toBe('');
  });

  test('previewStyle、title、icon 应该渲染', () => {
    const wrapper = render(
      <UploadSelector previewStyle={{ width: 200, height: 200 }} title="title" icon="icon" max={2} filesCount={1} />,
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
