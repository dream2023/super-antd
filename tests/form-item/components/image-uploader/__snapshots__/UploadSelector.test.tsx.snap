// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadSelector previewStyle、title、icon 应该渲染 1`] = `
<div>
  <div
    class="super-antd-center"
    style="width: 200px; height: 200px;"
  >
    <div>
      <div>
        icon
      </div>
      <div>
        title
      </div>
    </div>
  </div>
</div>
`;
